### ISSUE TITLE
`showManageSubscriptions` and `beginRefundRequest`: swiftUI view modifiers

### ISSUE THREAD

**Issue by aboedo** [RevenueCat/purchases-ios#1073]
For SwiftUI apps, we should provide view modifiers to make it easy to use `showManageSubscriptions` and `beginRefundRequest`. 

We can use `#if canImport(SwiftUI)` to only declare these when SwiftUI types are available.

For reference, here are the view modifiers declared by StoreKit. 

```swift
// Available when SwiftUI is imported with StoreKit
@available(iOS 15.0, *)
@available(macOS, unavailable)
@available(tvOS, unavailable)
@available(watchOS, unavailable)
extension View {

    public func manageSubscriptionsSheet(isPresented: Binding<Bool>) -> some View

}

// Available when SwiftUI is imported with StoreKit
@available(iOS 15.0, *)
@available(macOS, unavailable)
@available(tvOS, unavailable)
@available(watchOS, unavailable)
extension View {

    public func refundRequestSheet(for transactionID: UInt64, isPresented: Binding<Bool>, onDismiss: ((Result<Transaction.RefundRequestStatus, Transaction.RefundRequestError>) -> ())? = nil) -> some View

}

```

**Comment by NachoSoto** [comment]
I imagine we will grow SwiftUI extensions a lot over time. We might want to offer them separately eventually, or maybe now?
There's something to be said about thin frameworks, and if a user doesn't need SwiftUI it doesn't make sense to include it for everyone IMO.

**Comment by Juanpe** [comment]
🤔 Good point.

My two cents. Maybe not now, but as you said, there could be another UI related framework to isolate the Core layer from the UI.

**Comment by aboedo** [comment]
Yeah, I think it'd be nice and easy to set up as a separate pod and swift package. 
It'd also allow us to provide other extensions for SwiftUI to make life a lot easier for devs

**Comment by Zaprogramiacz** [comment]
Hey 👋

I spotted the thread when I was looking here and there. It has the `you can do this` tag, so is it something that I can contribute to or the topic is on hold?

**Comment by aboedo** [comment]
Hey @Zaprogramiacz! 👋 

You are absolutely welcome to contribute to this! We're more than happy to review and to assist with any questions you have along the way 💪

**Comment by Zaprogramiacz** [comment]
Oh great 💪🏻

Do you have any documentation, task with AC or desired dreamed solution? I don't have an access to the link from the first post and I bet that I might find there useful info.

In the thread I saw that last idea was to setup a separate pod and swift package with SwiftUI extensions. Is it still the main idea that should be followed?

**Comment by aboedo** [comment]
Hey, sorry for the late reply. Yeah, that's still the main idea, to have a separate SPM package and maybe also a pod, with just the SwiftUI extensions

**Comment by suprie** [comment]
I think this issue already resolved in https://github.com/RevenueCat/purchases-ios/blame/main/Sources/Support/ManageSubscriptionsHelper.swift#L34

**Comment by marcpalmer** [comment]
@surprie this is actually not the case that I can see.

There are other APIs missing in RC that I believe are required. Specifically to the issue here about view modifiers — there are no view modifiers in `ManageSubscriptionsHelper`.

This is problematic because of the deeper issue: StoreKit has problems with showing purchase and manage subscription sheets etc. in the correct window scene. If have e.g. a multi-window iPad or Catalyst app, the old APIs that don't take a `UIWindowScene` parameter use a non-deterministic window based on the window state.

This is why they added the new APIs like `showManageSubscriptions(in:, subscriptionGroupID:)` [Docs](https://developer.apple.com/documentation/storekit/appstore/showmanagesubscriptions(in:subscriptiongroupid:)) and you have to pass in a UIWindowScene.

This is actually painful to achieve in SwiftUI as you are not given direct access to the UIWindowScene as that is not a SwiftUI API — **which is why the SwiftUI modifiers like `manageSubscriptionsSheet(isPresented:)`** are required so that difficult detail can be hidden. After all, SwiftUI knows what Window Scene the view is currently in.

**Furthermore**, I would like to ask that we also get a `subscriptionGroupID` argument to all the RC APIs for showing the manage subscriptions UI because in recent iOS releases this provides a streamlined management UI that focuses on just the current subscription group they have and not the more general case of possibly multiple subscription groups. This was covered in their WWDC session. See [WWDC session](https://developer.apple.com/videos/play/wwdc2023/10140?time=608)
